deploy/helm/kubecf/templates/_config.tpl

```diff
@@ -0,0 +1,290 @@
+|
+| Lookup $path under $context and return either the value found, or the empty string.
+|
+| $context is either is an object (or array). It normally starts out as $.kubecf.config
+| or $.kubecf.manifest, but moves down the tree as _lookup calls itself recursively.
+|
+| $path is a list of properties to look up successively, e.g. ("stacks" $stack "releases").
+|
+| When the found value is `nil`, we still return the empty string and not a stringified
+| nil (which would be "<no data>"), because helm prints it as an empty string, but
+| otherwise treats as a non-empty string for all other purposes.
+|
+| $kubecf.retval is also set to the found entry in case the caller needs an object and
+| not just a string. This can also be used to disambiguate between nil and the empty
+| string (which can mean: not found).
+|
+| When $context is an array, then _lookup will look for an array element that has a "name"
+| property matching the first element of $path.
+|
+| Example: Looking for "instance_groups.api.jobs.cloud_controller_ng" in $.kubecf.manifest
+| is equivalent to the "/instance_groups/name=api/jobs/name=cloud_controller_ng" path in
+| JSON-patch.
+|
+| If the first element of $path for a list lookup contains a '=', the left side of specifies
+| the property to look for, in case it is not "name", e.g. "foo/os=linux/description".
++-----------------------------------------------------------------------------------------
+| Not implemented:
+|
+| When the full $path cannot be found, and the $.kubecf.required variable is true, then
+| _lookup should throw an error instead of returning an empty string. It could use
+| $.kubecf.query to provide a helpful error message. With the introduction of a
+| validation schema, this may no longer be required.
+==========================================================================================
+*/}}
+{{- define "_config._lookup" }}
+  {{- $kubecf := index . 0 }}
+  {{- $context := index . 1 }}
+  {{- $path := slice . 2 }}
+  {{- $key := (first $path) }}
+
+  {{- if kindIs "slice" $context }}
+    {{- $name := "name" }}
+    {{- if contains "=" $key }}
+      {{- $keyList := splitList "=" $key }}
+      {{- $name = first $keyList }}
+      {{- $key = index $keyList 1 }}
+    {{- end }}
+    {{- $_ := set $kubecf "retval" nil }}
+    {{- range $context }}
+      {{- if eq (index . $name) $key }}
+        {{- $_ := set $kubecf "retval" . }}
+      {{- end }}
+    {{- end }}
+  {{- else }}
+    {{- $_ := set $kubecf "retval" (index $context $key) }}
+  {{- end }}
+
+  {{- if $kubecf.retval }}
+    {{- if gt (len $path) 1 }}
+      {{- include "_config._lookup" (concat (list $kubecf $kubecf.retval) (rest $path)) }}
+    {{- else }}
+      {{- $kubecf.retval }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config.condition (list $ $condition)
```

Comment:
While I don't want to include general tests for the generated manifest in this PR, I do think we need a test for `_config.condition` to make sure it works as expected. I'll figure something out.

Reply:
feel free to use go to build a suite of unit tests for this; you should be able to use helm's libraries to automate some template testing

Reply:
I really want to avoid writing tests for kubecf in Go; it is too much work. 😸 

In 1dc62f0 I've created a special helm chart to call the template functions and verify they return the results I expect, and `{{fail}}` when they don't. It is testing the code in their natural environment, and tests are much more concise this way. I'm pretty happy with how it turned out; check it out! (May still need some cleanup and doc updates on Monday).